spec/models/piece_spec.rb

```diff
@@ -4,13 +4,140 @@
+      white_pawn = FactoryGirl.create(:pawn, :white, game_id: game.id)
+
+      expect(white_pawn.capturable?(4, 5)).to eq false
+    end
+
+    it 'return false if specified coordinates occupied by same color piece' do
+      game = empty_game
+      white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 3,
+        y_position: 0,
+        captured: false
+      )
+      other_white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 1,
+        captured: false
+      )
+
+      expect(other_white_pawn.capturable?(
+               white_pawn.x_position,
+               white_pawn.y_position
+      )).to eq false
+    end
+
+    it 'return true if specified coordinates occupied by target piece' do
+      game = empty_game
+      black_rook = FactoryGirl.create(
+        :rook,
+        :black,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 0,
+        captured: false
+      )
+      white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 1,
+        captured: false
+      )
+
+      expect(white_pawn.capturable?(
+               black_rook.x_position,
+               black_rook.y_position
+      )).to eq true
+    end
+  end
+
+  describe 'capture!' do
+    it 'update target piece attributes' do
+      game = empty_game
+      black_rook = FactoryGirl.create(
+        :rook,
+        :black,
+        game_id: game.id,
+        x_position: 3,
+        y_position: 1,
+        captured: false
+      )
+      white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 1,
+        captured: false
+      )
+
+      black_rook.capture!(white_pawn.x_position, white_pawn.y_position)
+
+      white_pawn.reload
+
+      expect(white_pawn.x_position).to eq nil
+      expect(white_pawn.y_position).to eq nil
+      expect(white_pawn.captured).to eq true
+    end
+  end
+
+  describe 'friendly_piece?' do
+    it 'return true if target piece have the same color' do
+      game = empty_game
+      white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 3,
+        y_position: 0,
+        captured: false
+      )
+      other_white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 1,
+        captured: false
+      )
+
+      expect(other_white_pawn.friendly_piece?(
+               white_pawn.x_position,
+               white_pawn.y_position
+      )).to eq true
+    end
+  end
+
+  # before create a new game, destory all pieces pre-populated from game model
+  def empty_game
+    game = FactoryGirl.create(:game)
+    game.pieces.destroy_all
+    game.reload
+    game
+  end
```

Comment:
I always prefer readability over a marginal increase of DRY in the test, since I believe tests should act like a documentation for how the application functions. This is especially true for this case. Since we already use a callback to populate the chess board (which hides a lot of details already), if we use `let` or `before` it will hide further more details and make the tests even harder to understand and make refactoring in the future much more difficult. 

This is just my option on this subject, if you really want to use `let` or `before` I can implement it also.